### ISSUE TITLE
MultiSearch response should include generic aggregations type

### ISSUE THREAD

**Issue by hoiekim** [elastic/elasticsearch-js#1937]
## 🐛 Bug Report

Hi maintainers,

Regarding this pull request #1596 (Aggregations should be a generic in responses), I believe there was missing changes in this line
https://github.com/elastic/elasticsearch-js/blob/2375059aa36281a5b7b004e3d0838a96ddf04238/src/api/types.ts#L782
Note that `TAggregations` is not passed in `MsearchMultiSearchResult`. This results in typescript not being able to recognize aggregations type in the response of `Client.msearch<TDocument, TAggregations>`.

## To Reproduce

```js
import { Client } from "@elastic/elasticsearch";
import { AggregationsTermsAggregateBase } from "@elastic/elasticsearch/lib/api/types";

const client = new Client({ node: "http://localhost:9200" });

const reproduce = async () => {
  interface Doc {
    foo: string;
  }

  interface Aggregations {
    unique: AggregationsTermsAggregateBase<{ key: string }>;
  }

  const response = await client.msearch<Doc, Aggregations>({
    index: "foo",
    searches: [
      {},
      {
        query: { match_all: {} },
        aggregations: { unique: { terms: { field: "foo" } } }
      }
    ]
  });

  response.responses.map((e) => {
    if ("error" in r) return
    return e.aggregations?.unique.buckets;
    //                            ^^^^^^^
    // Property 'buckets' does not exist on type 'AggregationsAggregate'.
    // Property 'buckets' does not exist on type 'AggregationsCardinalityAggregate'.
  });
};
```

## Expected behavior

Should not throw compile errors

## Your Environment

- *node version*: 18.15.0
- `@elastic/elasticsearch` *version*: 8.8.1
- `typescript` *version*: 4.5.5
- *os*: MacOS

**Comment by JoshMock** [comment]
Hi @hoiekim, sorry for the long delay.

I spent the day looking at this and did find one type, `MsearchMultiSearchResult`, where it can be improved to include a generic aggregation type.

However, in the case of `MsearchMultiSearchItem`, it already inherits a `ResponseBody` that includesa generic aggregation type, so there's nothing I can do there. The reason you get a TypeScript message is because `response.responses` is a `MsearchResponseItem`, which itself is a union:

https://github.com/elastic/elasticsearch-js/blob/99bcff02d5edf0e9200f3a36baf6df9f32aa6bf8/src/api/types.ts#L785

In TypeScript, unions create uncertainty about what properties may be available on an object. In this case, `ErrorResponseBase` does not have a `buckets` property, which is why TypeScript raises the error.

You can reproduce this in a simple example here:

```typescript
class FooBar {
  title: string
}

class BazBat {
  count: number
}

type FixBat = FooBar | BazBat

function getFixBat(): FixBat {
  if (Math.random() > 0.5) {
    return { title: 'foo' }
  } else {
    return { count: 10 }
  }
}

const f = getFixBat()
console.log(f.count)
// ^^ Property 'count' does not exist on type 'FixBat'. typescript (2339) [238, 15]
```	

Long story short: I will try to get the `MsearchMultiSearchResult` fix out for 8.11, but it won't resolve your error. There's not much we can do about that, unfortunately, other than telling TypeScript to ignore that line.
